crates/astria-conductor/src/executor.rs

```diff
@@ -273,49 +324,32 @@ impl<C: ExecutionClientV1Alpha1> Executor<C> {
                 // execute_block will check if our namespace has txs; if so, it'll return the
                 // resulting execution block hash, otherwise None
                 let Some(execution_block_hash) = self
-                    .execute_block(block)
+                    .execute_block(block.clone())
                     .await
                     .wrap_err("failed to execute block")?
                 else {
                     // no txs for our namespace, nothing to do
-                    debug!("execute_block returned None; skipping finalize_block");
+                    debug!("execute_block returned None; skipping call_update_commitment_state");
                     return Ok(());
                 };
 
-                // finalize the block after it's been executed
-                self.finalize_block(execution_block_hash, sequencer_block_hash)
-                    .await?;
+                // after execution, set SOFT as this block. FIRM stays the same.
+                let timestamp = convert_tendermint_to_prost_timestamp(block.header.time)?;
+                let firm = self.commitment_state.firm.clone().unwrap();
+                let soft = Block {
```

Comment:
I think the `firm` and `soft` commits here need to be reversed.
Updating the commit state to finalize a block we just executed should set `firm` to that block and `soft` should stay at whatever it was unless `firm` and `soft` become the same block.